packages/langium/src/lsp/completion/completion-provider.ts

```diff
@@ -184,19 +156,20 @@ export class DefaultCompletionProvider implements CompletionProvider {
             }
         };
 
-        await Promise.all(
-            stream(beforeFeatures)
-                .distinct(distinctionFunction)
-                .map(e => this.completionForContexts(contexts, e, acceptor))
-        );
-
-        if (reparse) {
+        const completedFeatures: NextFeature[] = [];
+        for (const context of contexts) {
```

Comment:
It would be nice to have a way to stop processing other contextes inside a called complitionFor(). Or at least move this for loop to a separate protected method

Reply:
I've added a `continueCompletion` method, that's supposed to return whether to continue with the next context.